sycl/include/sycl/ext/oneapi/experimental/graph.hpp

```diff
@@ -142,6 +142,14 @@ class __SYCL_EXPORT node {
   /// Update the Range of this node if it is a kernel execution node
   template <int Dimensions> void update_range(range<Dimensions> executionRange);
 
+  /// Common Reference Semantics
+  friend bool operator==(const node &LHS, const node &RHS) {
+    return LHS.impl == RHS.impl;
+  }
+  friend bool operator!=(const node &LHS, const node &RHS) {
+    return LHS.impl != RHS.impl;
+  }
+
```

Comment:
```suggestion
  friend bool operator!=(const node &LHS, const node &RHS) {
    return !operator==(LHS, RHS);
  }

```

I'm not sure if this works with the hidden friend functions, but if it does I think it would be preferable to repeating logic for `!=`.

Reply:
I updated it to this syntax and it seems to work :+1: